### PR TITLE
Dev to test

### PR DESCRIPTION
**Pull request by doronmak**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import uvicorn
+import config
+from app import app, first_app_init
+
+if __name__ == "__main__":
+    first_app_init()
+    uvicorn.run(app, host=config.api_host, port=config.port, log_level=config.log_level)
```
Comment (bengabay11) [inline main.py:7]: add future support for `ssl`

File: main.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import uvicorn
+import config
+from app import app, first_app_init
+
+if __name__ == "__main__":
+    first_app_init()
```
Comment (bengabay11) [inline main.py:6]: so you create the instance of the app in other file, and then import it?
why not move `first_app_init` to this file:
```pycon
app = first_app_init()
uvicorn.run(app, host=config.api_host, port=config.port, log_level=config.log_level)
```

**Review by bengabay11 — CHANGES_REQUESTED** [review]
- create a dedicated folder for the source code
- how do you run the code in production?
- if some part of the code is with python annotations I recommend you to add the annotations in all your code

File: app.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+@app.get('/restaurants')
+def get_all_restaurant():
+    rest_list = restaurants_db.get_all_restaurants()
+    return rest_list
+
+
+@app.post('/restaurant')
+async def post_restaurant(restaurant: Restaurant):
```
Comment (bengabay11) [inline app.py:43]:
```suggestion
async def post_restaurant(restaurant: Restaurant) -> JSONResponse:
```

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+uvicorn~=0.12.1
```
Comment (bengabay11) [inline requirements.txt:1]: you need dev requirements file. you can use `pipenv` to manage that.

File: app.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+app = FastAPI()
+
+restaurants_db = DB(db_type=config.db_type, db_user=config.db_user, db_password=config.db_password,
+                    db_host=config.db_host,
+                    db_name=config.db_name)
+
+csv_reader = CSVAnalyzer(config.Restaurant_file)
+
+
+class Restaurant(BaseModel):
```
[the remark this answers is not in the document]
Comment (doronmak) [reply app.py]: Done

File: app.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    return "Api is on fire!"
+
+
+@app.get('/restaurants')
+def get_all_restaurant():
+    rest_list = restaurants_db.get_all_restaurants()
+    return rest_list
+
+
+@app.post('/restaurant')
```
[the remark this answers is not in the document]
Comment (doronmak) [reply app.py]: Done

File: app.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+async def post_restaurant(restaurant: Restaurant):
+    restaurants_db.add_new_restaurant(restaurant_name=restaurant.Name, restaurant_type=restaurant.Type, restaurant_phone=restaurant.Phone,
+                                      restaurant_location=restaurant.Location)
+    return "post new restaurant"
```
[the remark this answers is not in the document]
Comment (doronmak) [reply app.py]: Done

File: config.py [hunk 8]
```diff
@@ -0,0 +1,12 @@
+api_host = "localhost"
```
[the remark this answers is not in the document]
Comment (doronmak) [reply config.py]: Done

File: database/models.py [hunk 9]
```diff
@@ -0,0 +1,13 @@
+from sqlalchemy import Column, String, Integer
+from sqlalchemy.ext.declarative import declarative_base
+
+Base = declarative_base()
+
+
+class Restaurants(Base):
```
[the remark this answers is not in the document]
Comment (doronmak) [reply database/models.py]: Done

File: requirements.txt [hunk 10]
```diff
@@ -0,0 +1,4 @@
...
+fastapi~=0.61.1
+SQLAlchemy~=1.3.19
+pydantic~=1.6.1
```
[the remark this answers is not in the document]
Comment (doronmak) [reply requirements.txt]: ok

**Pull request closed without merging**